pkg/watcher/skr_webhook_manifest_manager.go

```diff
@@ -237,13 +246,24 @@ func (m *SKRWebhookManifestManager) getUnstructuredResourcesConfig(ctx context.C
 		cpuResLimit:     m.config.SkrWebhookCPULimits,
 		memResLimit:     m.config.SkrWebhookMemoryLimits,
 		skrWatcherImage: m.config.SkrWatcherImage,
-		caCert:          tlsSecret.Data[caCertKey],
+		caCert:          gatewaySecret.Data[caCertKey],
 		tlsCert:         tlsSecret.Data[tlsCertKey],
 		tlsKey:          tlsSecret.Data[tlsPrivateKeyKey],
 		remoteNs:        remoteNs,
 	}, nil
 }
 
+func getCertificateSecret(ctx context.Context, clnt client.Client, objKey client.ObjectKey) (*apicorev1.Secret, error) {
```

Comment:
all of objKey are: `Namespace: m.certificateConfig.IstioNamespace, Name: gatewaysecret.GatewaySecretName,`, right? why not design this function belongs to `CertificateManager` as `func (c *CertificateManager) getCertificateSecret(ctx context.Context)`

Reply:
I expected a function comment to explain the purpose of this secret.

Reply:
This function is used in 5 places. Sometimes for the gateway secret and sometimes for the watcher secret. Hence was made generic

Reply:
makes sense!